### PR TITLE
Port Compact Convolutional Transformers to backend agnostic

### PR DESCRIPTION
**Pull request by guillaumebaquiast**
Port Compact Convolutional Transformers to be backend agnostic. Works on TF, Jax, and Torch backends (at least on my machine).

Two notes:
* I used the same fix as in https://github.com/keras-team/keras-core/pull/663 to make the Position Embedding work (without the fix, it fails with Torch, and doesn't behave as expected on Jax and Tensorflow)
* The implementation of `StochasticDepth` was failing for Torch, due to what I believe was a typo from the original implementation.

### CODE REVIEW

File: examples/keras_io/tensorflow/vision/cct.py [hunk 1]
```diff
@@ -144,32 +139,77 @@ def call(self, images):
...
-            embed_layer = layers.Embedding(
-                input_dim=sequence_length, output_dim=projection_dim
+
+"""
+Positional embeddings are optional in CCT. If we want to use them, we can use
+the Layer defined below.
+"""
+
+
+class PositionEmbedding(keras.layers.Layer):
```
Comment (guillaumebaquiast) [inline examples/keras_io/tensorflow/vision/cct.py:159]: Same as in https://github.com/keras-team/keras-core/pull/663

File: examples/keras_io/tensorflow/vision/cct.py [hunk 2]
```diff
@@ -214,9 +255,9 @@ def __init__(self, drop_prop, **kwargs):
     def call(self, x, training=None):
         if training:
             keep_prob = 1 - self.drop_prob
-            shape = (tf.shape(x)[0],) + (1,) * (tf.shape(x).shape[0] - 1)
-            random_tensor = keep_prob + tf.random.uniform(shape, 0, 1)
-            random_tensor = tf.floor(random_tensor)
+            shape = (keras.ops.shape(x)[0],) + (1,) * (len(x.shape) - 1)
```
Comment (guillaumebaquiast) [inline examples/keras_io/tensorflow/vision/cct.py:258]: `(tf.shape(x).shape[0] - 1)` made Torch fail (error message was complaining about too many dimensions to the tensor). I believe it was a typo from the original implementation, which I replaced by `(len(x.shape) - 1)`

**Review by fchollet — COMMENTED** [review]
Thanks for the PR! The changes look good. Since the example is backend-agnostic, you can move it outside of the `tensorflow` folder (to `keras_io/vision`).

**Review by fchollet — APPROVED** [review]
LGTM, thank you!

**Pull request merged**